uniffi_bindgen/src/bindings/udl/gen_udl.rs

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
Heh, so, this is a litle odd, but I also added a UDL-generating bindings backend. That is: you can use this to *parse* a `ComponentInterface` from a Rust source file and then *generate* a corresponding `.udl` file. This was entirely in support of https://github.com/mozilla/application-services/pull/3876, because I didn't want to keep the interface and its doc in sync between two different files. Consider this a way to experiment with the parsing-rust-code stuff without necessarily committing to it right away - I was able to write my crate *as though* we had all this fun macro stuff up and running, then automagically generate a corresponding `.udl` file so I could use it with the current release version of UniFFI.

I expect we would not want to actually land this or maintain it long term.

Reply:
While that's very cool, I agree we probably don't want it - I was very confused above when I saw a .udl back-end :)